developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+
+#### Fixing the Struggles
+
+Once raised, use the following steps to capture and create **actionable** fix.
+
+1. Define the problem - End to end testing is only possible when the solution is deployed to environment in Azure
+2. Identify a potential solution - Ability to run automated end-to-end tests on the developer machine.
+3. Justify fixing the problem - The team has identified this issue as the root cause for 3 stories not completing within their sprint over the past 2 sprints.
+
+### Single IDE Instance
+
+Avoid requiring multiple instances of the IDE to running concurrently to run and debug the solution locally. This will negatively impact the above measure. Additionally, that negative impact is multiplied by each instance required.
+
+- Configuration steps * IDE instances
+- Build steps * IDE instances
+- Start/Debug steps * IDE instances
+- Stop steps * IDE instances
+- Run test steps * IDE instances
+- Documenting all of the above * IDE instances
+
+### Single Repository
+
+Splitting a solution across multiple repositories can negatively impact the above measures. This can also negatively impact other areas such as Pull Requests, Automated Testing, Continuous Integration, and Continuous Delivery. Similar to the IDE instances, the negative impact is multiplied by the number of repositories.
+
+- clone steps * # of repos
+- branching steps * # of repos
+- commit steps * # of repos
+- CI * # of repos
+- Pull request reviews & merges * # of repos
+
+#### Atomic Pull Requests
+
+When the solution is encapsulated within a single repository, it also allows pull requests to represent a change across multiple layers. This is especially helpful when a change requires changes to a shared contract between multiple components. For example, a story requires that an api endpoint is changed. With this strategy the api and web client could be updated with the same pull request. This avoids master being broken temporarily while waiting on dependent pull requests to merge.
+
+[`TODO: Add link to other documented benefits of consolidated repos within source control section`]
+
+### When to Split Into Multiple Repositories
+
+[`TODO: Add link to source control section if it has guidance on this subject`]
+
+(Typically try to have the repository align with the product backlog.)
+
+### Minimize Remote Dependencies for Local Development
+
+The fewer dependencies on components that cannot run a developer's machine translate to fewer steps required to get started. Therefore, fewer dependencies will positively impact the measures above.
+
+The following strategies can be used to reduce these dependencies
+
+#### Use an Emulator
+
+If available, emulators are implementations of technologies that are typically only available in cloud environments. A good example is the [CosmosDB emulator](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator).
+
+#### Feature Toggle a Mock or Different Technology

```

Comment:
I guess depending on scope of the abstracted element a feature toggle might be appropriate... for more narrow abstractions DI can be better and is worth a mention here.

Reply:
That was what i was trying to convey but, probably not doing it successfully. 

In my experience, you would use DI to inject a different implementation of a shared interface. However, that is also usually encapsulated within some type of factory logic where a decision is made based on some env config/setting. I was referring to that practice as feature toggling but, perhaps that leaves to much to interpretation (or is the wrong interpretation?). 

Maybe need to be more explicit? Thoughts?

Reply:
I hear you... when I saw "feature toggling" I read "feature flags" which is conceptually related to but distinct from DI (different scopes, different mechanisms, different rules of engagement, etc.). I'd recommend the word be a bit more explicit.

Reply:
Updated and provided example. let me know if this better represents what you were suggesting.